gtyped/codecs.go

```diff
@@ -0,0 +1,46 @@
+package gtyped
+
+import (
+	"fmt"
+
+	"github.com/lovoo/goka"
+)
+
+type GCodec[V any] interface {
+	GEncode(value V) (data []byte, err error)
+	GDecode(data []byte) (value V, err error)
+}
+
+type codecBridge[T any] struct {
+	c GCodec[T]
+}
+
+func (cb *codecBridge[T]) Encode(value interface{}) ([]byte, error) {
+	tVal, ok := value.(T)
+	var zero T
+	if !ok {
+		return nil, fmt.Errorf("unexpected type while encoding. Expected %T, got %T", zero, value)
+	}
+	return cb.c.GEncode(tVal)
+}
+
+func (cb *codecBridge[T]) Decode(data []byte) (interface{}, error) {
+	return cb.c.GDecode(data)
+}
+
+// CodecBridge converts a untyped (goka) codec to a typed codec
```

Comment:
From the signature, it takes a typed codec and returns the untyped codec.
The comment is the other way around?

Reply:
true that, changed the comment.